packages/cache/src/options.ts

```diff
@@ -15,3 +17,93 @@ export interface UploadOptions {
    */
   uploadChunkSize?: number
 }
+
+/**
+ * Options to control cache download
+ */
+export interface DownloadOptions {
+  /**
+   * Indicates whether to use the Azure Blob SDK to download caches
+   * that are stored on Azure Blob Storage to improve reliability and
+   * performance
+   *
+   * @default true
+   */
+  useAzureSdk?: boolean
+
+  /**
+   * Number of parallel downloads (this option only applies when using
+   * the Azure SDK)
+   *
+   * @default 8
+   */
+  downloadConcurrency?: number
+
+  /**
+   * Maximum time for each download request, in milliseconds (this
+   * option only applies when using the Azure SDK)
+   *
+   * @default 30000
+   */
+  timeoutInMs?: number
+}
+
+/**
+ * Returns a copy of the upload options with defaults filled in.
+ *
+ * @param copy the original upload options
+ */
+export function getUploadOptions(copy?: UploadOptions): UploadOptions {
+  const result: UploadOptions = {
+    uploadConcurrency: 4,
+    uploadChunkSize: 32 * 1024 * 1024
+  }
+
+  if (copy) {
+    if (typeof copy.uploadConcurrency === 'number') {
+      result.uploadConcurrency = copy.uploadConcurrency
+    }
+
+    if (typeof copy.uploadChunkSize === 'number') {
+      result.uploadChunkSize = copy.uploadChunkSize
+    }
+  }
+
+  core.debug(`Upload concurrency: ${result.uploadConcurrency}`)
+  core.debug(`Upload chunk size: ${result.uploadChunkSize}`)
+
+  return result
+}
+
+/**
+ * Returns a copy of the download options with defaults filled in.
+ *
+ * @param copy the original download options
+ */
+export function getDownloadOptions(copy?: DownloadOptions): DownloadOptions {
+  const result: DownloadOptions = {
+    useAzureSdk: true,
+    downloadConcurrency: 8,
+    timeoutInMs: 30000
+  }
+
+  if (copy) {
+    if (typeof copy.useAzureSdk === 'boolean') {
```

Comment:
any reason for checking these types? we are already in typescript, we should have good typechecking. I guess are these sent by customers and they can send anything they want? Why don't we validate them and fail if that's the case

Reply:
I think the purpose of this if condition is to check if the value is set (not undefined and not null).  So I think this is just an easier expression.  (This pattern is used elsewhere, such as https://github.com/actions/toolkit/blob/1a2c5929039617047c4b37b17b1c4cd6de0b8333/packages/glob/src/internal-glob-options-helper.ts#L15)